common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -220,6 +230,40 @@ def perform_destroy(self, instance):
         # This is not called with is_refund=True here because it is assumed the user has already been refunded.
         _process_revoke_and_unenroll_entitlement(instance)
 
+    def partial_update(self, request, *args, **kwargs):
+        entitlement_uuid = kwargs.get('uuid', None)
+
+        try:
+            entitlement = CourseEntitlement.objects.get(uuid=entitlement_uuid)
+        except CourseEntitlement.DoesNotExist:
+            return HttpResponseBadRequest(
+                u'Could not find entitlement {entitlement_uuid} to update'.format(
+                    entitlement_uuid=entitlement_uuid
+                )
+            )
+        support_details = request.data.pop('support_details', [])
+
+        for support_detail in support_details:
+            parsed_details = ast.literal_eval(support_detail)
+            parsed_details['entitlement'] = entitlement
+            parsed_details['support_user'] = request.user
+            unenrolled_run_id = parsed_details.get('unenrolled_run', None)
+            if unenrolled_run_id:
+                try:
+                    with transaction.atomic():
+                        unenrolled_run_course_key = CourseKey.from_string(unenrolled_run_id)
+                        CourseEnrollment.unenroll(entitlement.user, unenrolled_run_course_key, skip_refund=True)
+                        parsed_details['unenrolled_run'] = CourseOverview.objects.get(id=unenrolled_run_course_key)
+                except:
+                    return HttpResponseBadRequest(
+                        u'Could not unenroll user from course run {course_id}'.format(
```

Comment:
any chance we could add more detail about the unenroll failure? might be useful at somepoint.